<launch>

  <arg name="map_frame_id" default="map"/>
  <arg name="map_size_x" default="40"/>
  <arg name="map_size_y" default="40"/>
  <arg name="map_size_z" default="5"/>
  <arg name="map_x_origin" default="-20.0"/>
  <arg name="map_y_origin" default="-20.0"/>
  <arg name="map_z_origin" default="0.0"/>

  <node pkg ="param_env" name ="structure_map" type ="structure_map" output = "screen">    
    <!-- this is the map init position, not the odom-->

    <param name="map/x_size"     value="$(arg map_size_x)" />
    <param name="map/y_size"     value="$(arg map_size_y)" />
    <param name="map/z_size"     value="$(arg map_size_z)" />
    <param name="map/x_origin"   value="$(arg map_x_origin)"/>
    <param name="map/y_origin"   value="$(arg map_y_origin)"/>
    <param name="map/z_origin"   value="$(arg map_z_origin)"/>

    <param name="map/resolution" value="0.1"/>        
    <param name="map/frame_id"     value="$(arg map_frame_id)" />

    <!-- total obstacle ratios -->
    <param name="params/cylinder_ratio" value="0.02" type="double"/>
    <param name="params/circle_ratio"   value="0.02" type="double"/>
    <param name="params/gate_ratio"     value="0.02" type="double"/>
    <param name="params/ellip_ratio"    value="0.02" type="double"/>
    <param name="params/poly_ratio"     value="0.02" type="double"/>

    <param name="params/w1" value="0.1"/>
    <param name="params/w2" value="0.8"/>          
    <param name="params/w3" value="1.2"/>
    <param name="params/w4" value="3.0"/>
    
    <!-- add gaussion noise with param/w1 range on it -->
    <param name="params/add_noise" value="true"/>
 

    <!-- to generate map dataset in this folder -->
    <param name="dataset/save_map"    value="true"/>
    <param name="dataset/start_index" value="5080900"/>
    <param name="dataset/path"        value="$(find param_env)/dataset/"/>


  </node>


  <arg name="rviz_config" default="$(find param_env)/launch/default.rviz"/>
  <node pkg="rviz"
    type="rviz"
    name="rviz"
    args="-d $(arg rviz_config)"/>


</launch>
